hue/hue.sh

```diff
@@ -24,127 +29,142 @@ function update_apt_get() {
+local new_mysql_settings='engine=mysql$1host=127.0.0.1$2port=3306$3user=hue$4password=hue-password'
+local mysql_password='root-password'
+local hue_password='hue-password'
 # Install hue
-update_apt_get
-apt-get install -t jessie-backports hue -y
+apt-get install -t jessie-backports hue -y || err "Failed to install hue"
 
 # Stop hue
-systemctl stop hue
-
-cat > core-site-patch.xml <<EOF
-<property>
-  <name>hadoop.proxyuser.hue.hosts</name>
-  <value>*</value>
-</property>
-<property>
-  <name>hadoop.proxyuser.hue.groups</name>
-  <value>*</value>
-</property>
-EOF
-sed -i '/<\/configuration>/e cat core-site-patch.xml' \
-   /etc/hadoop/conf/core-site.xml
+systemctl stop hue || err "Hue stop action not performed"
 
+bdconfig set_property \
+  --configuration_file 'core-site-patch.xml' \
+  --name 'hadoop.proxyuser.hue.hosts' --value '*' \
+  --clobber \
+  || err 'Unable to set hadoop.proxyuser.hue.hosts'
 
-cat > hdfs-site-patch.xml <<EOF
-<property>
-  <name>dfs.webhdfs.enabled</name>
-  <value>true</value>
-</property>
+bdconfig set_property \
+  --configuration_file 'core-site-patch.xml' \
+  --name 'hadoop.proxyuser.hue.groups' --value '*' \
+  --clobber \
+  || err 'Unable to set hadoop.proxyuser.hue.groups'
 
-<property>
-  <name>hadoop.proxyuser.hue.hosts</name>
-  <value>*</value>
-</property>
-
-<property>
-  <name>hadoop.proxyuser.hue.groups</name>
-  <value>*</value>
-</property>
-EOF
+sed -i '/<\/configuration>/e cat core-site-patch.xml' \
+  /etc/hadoop/conf/core-site.xml
+
+bdconfig set_property \
+  --configuration_file 'hdfs-site-patch.xml' \
+  --name 'dfs.webhdfs.enabled' --value 'true' \
+  --clobber \
+  || err 'Unable to set dfs.webhdfs.enabled'
+
+bdconfig set_property \
+  --configuration_file 'hdfs-site-patch.xml' \
+  --name 'hadoop.proxyuser.hue.hosts' --value '*' \
+  --clobber \
+  || err 'Unable to set hadoop.proxyuser.hue.hosts'
+
+bdconfig set_property \
+  --configuration_file 'hdfs-site-patch.xml' \
+  --name 'hadoop.proxyuser.hue.groups' --value '*' \
+  --clobber \
+  || err 'nable to set hadoop.proxyuser.hue.groups'
 
 sed -i '/<\/configuration>/e cat hdfs-site-patch.xml' \
-     /etc/hadoop/conf/hdfs-site.xml
+  /etc/hadoop/conf/hdfs-site.xml
 
 cat >> /etc/hue/conf/hue.ini <<EOF
-  # Defaults to $HADOOP_MR1_HOME or /usr/lib/hadoop-0.20-mapreduce
+  # Defaults to ${HADOOP_MR1_HOME} or /usr/lib/hadoop-0.20-mapreduce
   hadoop_mapred_home=/usr/lib/hadoop-mapreduce
 EOF
 
 # If oozie installed, add hue as proxy user for oozie
 if [[ -f /etc/oozie/conf/oozie-site.xml ]];
 then
-  cat > oozie-site-patch.xml <<EOF
-<property>
-  <name>oozie.service.ProxyUserService.proxyuser.hue.hosts</name>
-  <value>*</value>
-</property>
-
-<property>
-  <name>oozie.service.ProxyUserService.proxyuser.hue.groups</name>
-  <value>*</value>
-</property>
-EOF
+  bdconfig set_property \
+    --configuration_file 'oozie-site-patch.xml' \
```

Comment:
Same thing here -- let's avoid using this patch file. If this fails halfway through, cluster creation will fail anyway.